src/apps/service/apps.clj

```diff
@@ -76,7 +77,9 @@
 (defn get-app-categories
   [user params]
   (let [client (get-apps-client user "type=apps")]
-    {:categories (transaction (.listAppCategories client params))}))
+    {:categories (transaction
+                   (walk/prewalk (fn [x] (if (or (future? x) (delay? x)) (deref x) x))
```

Comment:
This makes it so any futures or delays are deref'd before the transaction ends. This is to ensure that any queries therein have started before the transaction ends, and also to replace their values with the non-delay/non-future version. Basically, force everything to have run before ending the transaction.